### PR TITLE
Support SIZEOF, CONTAINSKEY, CONTAINSVALUE, comparing to fields

### PR DESCRIPTION
**Pull request by hbxie**
Addresses #12.

Needs new release of bullet-core.

filed name is a string without quotes. String value is a string with quotes(double or single).

Support the following:

// These 'a' are values.
SELECT * FROM STREAM(3, TIME) WHERE aaa CONTAINSKEY (1, 3, 'a')
SELECT * FROM STREAM(3, TIME) WHERE aaa NOT CONTAINSKEY (1, 3, 'a')
SELECT * FROM STREAM(3, TIME) WHERE aaa CONTAINSVALUE (1, 3, 'a')

// this 'a' is a filed name
SELECT * FROM STREAM(3, TIME) WHERE aaa NOT CONTAINSVALUE (1, 3, a) 

SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa)=1
SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa)!=1
SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa)<>1
SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa) IS DISTINCT FROM 1
SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa) IS NOT DISTINCT FROM 1
SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa) IN (1, 2)
SELECT * FROM STREAM(3, TIME) WHERE SIZEOF(aaa) NOT IN (1, 2)

### CODE REVIEW

File: src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java [hunk 1]
```diff
@@ -85,17 +93,43 @@ protected Clause visitLogicalBinaryExpression(LogicalBinaryExpression node, Void
 
         @Override
         protected Clause visitComparisonExpression(ComparisonExpression node, Void context) {
-            return createFilterClause(node.getOperation(), node.getLeft().toFormatlessString(), node.getRight().toFormatlessString());
+            Expression left = node.getLeft();
+            Operation op = node.getOperation();
+            boolean isNotExpression = false;
+            if (left instanceof ReferenceWithFunction) {
+                if (op != EQUALS && op != NOT_EQUALS) {
```
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java]: I know we don't support SIZEOF(*) with >, < but is that possible to do? We need changes to bullet-core as well.

File: src/main/java/com/yahoo/bullet/bql/tree/ContainsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+public class ContainsPredicate extends Expression {
+    private final Expression value;
+    private final Expression valueList;
+    private final Operation operation;
+
+    /**
+     * Constructor that requires a {@link Operation} operation, an {@link Expression} value and an {@link Expression} valueList.
+     *
+     * @param operation      A {@link Operation}.
```
[the remark this answers is not in the document]
Comment (hbxie) [reply src/main/java/com/yahoo/bullet/bql/tree/ContainsPredicate.java]: Done.

File: src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java (hunk 1 above)
Comment (akshaisarma) [reply src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java]: Ah ok, maybe we should rename SIZEOF to SIZEOFEQUALS or something to be explicit

Comment (hbxie) [reply src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java]: I guess we can leave SIZEOF here. And we may extend it to support >, < and other operations and don't need to change the gramma once we do the change in bullet-core.

Comment (akshaisarma) [reply src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java]: I meant in bullet-core actually and the only way I see to support SIZE OF > etc is to allow another op to passed with SIZEOF, which I am not sure we'll be supporting anytime soon. Ok, let's leave it for now

**Pull request merged**